File: Bipin/013_array_rotation/array_rotation_test.go
package main

import (
	"reflect"
	"testing"
)

func TestRotateArray(t *testing.T) {
	// Test case 1: Non-empty array with positive rotation
	nums := []int{1, 2, 3, 4, 5}
	k := 2
	RotateArray(nums, k)
	expected := []int{4, 5, 1, 2, 3}
	if !reflect.DeepEqual(nums, expected) {
		t.Errorf("Test Case 1 Failed: expected %v, got %v", expected, nums)
	}

	// Test case 2: Rotation equal to array length
	nums = []int{1, 2, 3, 4, 5}
	k = 5
	RotateArray(nums, k)
	expected = []int{1, 2, 3, 4, 5}
	if !reflect.DeepEqual(nums, expected) {
		t.Errorf("Test Case 2 Failed: expected %v, got %v", expected, nums)
	}

	// Test case 3: Rotation greater than array length
	nums = []int{1, 2, 3, 4, 5}
	k = 7
	RotateArray(nums, k)
	expected = []int{4, 5, 1, 2, 3}
	if !reflect.DeepEqual(nums, expected) {
		t.Errorf("Test Case 3 Failed: expected %v, got %v", expected, nums)
	}

	// Test case 4: Empty array
	nums = []int{}
	k = 3
	RotateArray(nums, k)
	expected = []int{}
	if !reflect.DeepEqual(nums, expected) {
		t.Errorf("Test Case 4 Failed: expected %v, got %v", expected, nums)
	}
}

File: prakash/13_ArrayRotation/Arrayrotation.test.js
const rotateArray = require("./Arrayrotation");
describe("rotateArray", () => {
  test("Test with a non-empty array and positive rotation", () => {
    const nums = [1, 2, 3, 4, 5];
    rotateArray(nums, 2);
    expect(nums).toEqual([4, 5, 1, 2, 3]);
  });

  test("Test with rotation equal to array length", () => {
    const nums = [1, 2, 3, 4, 5];
    rotateArray(nums, 5);
    expect(nums).toEqual([1, 2, 3, 4, 5]);
  });

  test("Test with rotation greater than array length", () => {
    const nums = [1, 2, 3, 4, 5];
    rotateArray(nums, 7); // Equivalent to rotating by 2 steps
    expect(nums).toEqual([4, 5, 1, 2, 3]);
  });

  test("Test with an empty array", () => {
    const nums = [];
    rotateArray(nums, 3);
    expect(nums).toEqual([]);
  });
});
